src/tflite_cpu.cpp

```diff
@@ -397,15 +397,26 @@ class MLModelServiceTFLite : public vsdk::MLModelService,
             state->metadata.inputs.emplace_back(std::move(input_info));
         }
 
-        // As above, but for output tensors.
-        //
         // NOTE: The tflite C API docs state that information about
         // output tensors may not be available until after one round
-        // of inference. We are ignoring that guidance for now per the
-        // unknowns about how metadata will be handled in the future.
+        // of inference. We do a best effort inference on all zero
+        // inputs to try to account for this.
+        for (decltype(num_input_tensors) i = 0; i != num_input_tensors; ++i) {
+            auto* const tensor = TfLiteInterpreterGetInputTensor(state->interpreter.get(), i);
+            const auto tensor_size = TfLiteTensorByteSize(tensor);
+            const std::vector<unsigned char> zero_buffer(tensor_size, 0);
+            TfLiteTensorCopyFromBuffer(tensor, &zero_buffer[0], tensor_size);
+        }
+
+        const auto invoke_result = TfLiteInterpreterInvoke(state->interpreter.get());
+        if (invoke_result != TfLiteStatus::kTfLiteOk) {
+            // TODO: After C++ SDK 0.11.0 is released, use the new logging API.
```

Comment:
FWIW, I think logging was already [added in v0.10.0](https://github.com/viamrobotics/viam-cpp-sdk/releases/tag/releases%2Fv0.10.0). but I'm content keeping this as a TODO because I don't know how hard it is to upgrade from 0.9.0 to 0.10.0.

Reply:
Yes, it was. However, there are some reasons I don't want to move to 0.10.0 and prefer to wait for 0.11.0, which should be coming shortly.